documentation/docs/install-pmm/install-pmm-client/connect-database/mongodb.md

```diff
@@ -58,266 +64,245 @@ Run the example codes below in a `mongo` session to:
---tls-certificate-key-file-password=IFPASSWORDTOCERTISSET \
---tls-ca-file=PATHTOCACERT
---authentication-mechanism=AUTHENTICATION-MECHANISM
---authentication-database=AUTHENTICATION-DATABASE
-```
+    2. Fill in the required fields.
 
-where:
+    3. Click **Add service**.
 
-- `PATHTOCERT`: Path to TLS certificate file.
-- `IFPASSWORDTOCERTISSET`: Password for TLS certificate file.
-- `PATHTOCACERT`: Path to certificate authority file.
-- `AUTHENTICATION-MECHANISM`: Authentication mechanism. Default is empty. Use `MONGODB-X509` for SSL certificates.
-- `AUTHENTICATION-DATABASE`: Authentication database. Default is empty. Use `$external` for SSL certificates.
+    ![!](../../../images/PMM_Add_Instance_MongoDB.jpg)
 
-## Check the service
+=== "Add service via CLI"
 
-### From the user interface
+    Use `pmm-admin` to add the database server as a service using one of these example commands:
 
-To check the service from the UI:
-{.power-number}
+    === "Basic MongoDB instance"
+        ```sh
+        pmm-admin add mongodb \
+        --username=pmm \
+        --password=your_secure_password
+        ```
 
-1. Select  **PMM Configuration > Inventory > MongoDB**.
+    === "Sharded cluster component"
+        ```sh
+        pmm-admin add mongodb \
+        --username=pmm \
+        --password=your_secure_password \
+        --cluster=my_cluster_name \
+        --replication-set=rs1  # Optional: specify replication set name
+        ```
 
-2. Enter or select values for the fields.
+    === "SSL/TLS secured MongoDB"
+        ```sh
+        pmm-admin add mongodb \
+        --username=pmm \
+        --password=your_secure_password \
+        --tls \
+        --tls-certificate-key-file=/path/to/client.pem \
+        --tls-certificate-key-file-password=cert_password \  # If certificate has password
+        --tls-ca-file=/path/to/ca.pem \
+        --authentication-mechanism=MONGODB-X509 \  # For X.509 authentication
+        --authentication-database=$external      # For X.509 authentication
+        ```
 
-3. Click **Add service**.
+    When successful, PMM Client will print `MongoDB Service added` with the service's ID and name. Use the `--environment` and `--custom-labels` options to set tags for the service to help identify them.
 
-![!](../../../images/PMM_Add_Instance_MongoDB_TLS.jpg)
+    !!! hint alert alert-success "Tips"
+        - When adding nodes to a sharded cluster, ensure to add each node separately using the `--cluster mycluster` option. This allows the [MongoDB Cluster Summary](../../../reference/dashboards/dashboard-mongodb-cluster-summary.md) dashboard to populate correctly. 
+        - You can also use the `--replication-set` option to specify a replication set, although they are automatically detected. For instance, you can use `--replication-set config` for your config servers; `--replication-set rs1` for your servers in the first replica set, `--replication-set rs2` for your servers in the second replica set, and so on.
```

Comment:
it's not really automatically detected, we have some metrics which has information about replicasets, but in other metrics we rely on preset value. autodetection isn't implemented yet.

Reply:
can you give me an example of such metric? I have never ever used the --replication-set argument when registering agents and still dashboards seem to look fine unless I am missing something

Reply:
I see that we are using values from 
`label_values(mongodb_mongod_replset_my_state{environment=~"$environment",cluster=~"$cluster"}, set)` and it's not the same value that metrics get from --replication-set flag. it means that `--replication-set` flag is ignored in our dashboards.